storage/reads/aggregate_resultset.go

```diff
@@ -83,22 +83,19 @@ func convertNsecs(nsecs int64) values.Duration {
 	return values.MakeDuration(nsecs, 0, negative)
 }
 
-func (r *windowAggregateResultSet) createCursor(seriesRow SeriesRow) (cursors.Cursor, error) {
-	agg := r.req.Aggregate[0]
-	every := r.req.WindowEvery
-	offset := r.req.Offset
-	cursor := r.arrayCursors.createCursor(seriesRow)
-
+func GetWindow(req *datatypes.ReadWindowAggregateRequest) (interval.Window, error) {
 	var everyDur values.Duration
 	var offsetDur values.Duration
 	var periodDur values.Duration
 
-	if r.req.Window != nil {
+	every := req.WindowEvery
+	offset := req.Offset
+	if req.Window != nil {
 		// assume window was passed in and translate protobuf window to execute.Window
-		everyDur = values.MakeDuration(r.req.Window.Every.Nsecs, r.req.Window.Every.Months, r.req.Window.Every.Negative)
-		periodDur = values.MakeDuration(r.req.Window.Every.Nsecs, r.req.Window.Every.Months, r.req.Window.Every.Negative)
-		if r.req.Window.Offset != nil {
-			offsetDur = values.MakeDuration(r.req.Window.Offset.Nsecs, r.req.Window.Offset.Months, r.req.Window.Offset.Negative)
+		everyDur = values.MakeDuration(req.Window.Every.Nsecs, req.Window.Every.Months, req.Window.Every.Negative)
+		periodDur = values.MakeDuration(req.Window.Every.Nsecs, req.Window.Every.Months, req.Window.Every.Negative)
```

Comment:
everyDur and periodDur should have common initialization to enforce and highlight that every period should start every X and last for X time.  Code here is correct, but not clearly expressing the intent.

Reply:
Good catch - I want to just move these at the moment though, so going to leave it as-is.